zip-0244.rst

```diff
@@ -1,8 +1,413 @@
+```````````````````````````
+A BLAKE2b-256 hash of the field encoding of the JoinSplit signature
+belonging to the transaction.
+
+The personalization field of this hash is set to::
+
+  "ZTxAuthSprouHash"
+
+3: ``sapling_sigs_digest``
+```````````````````````````
+A BLAKE2b-256 hash of the field encoding of the Sapling signature
+of each Sapling spend description belonging to the transaction, followed by the
+field encoding of the binding signature.
+
+The personalization field of this hash is set to::
+
+  "ZTxAuthSapliHash"
+
+--------------------
+Block Header Changes
+--------------------
+
+// TODO: Need @str4d's help here.
+
+========================
+Reference implementation
+========================
+
+- https://github.com/zcash/librustzcash/pull/319/files
+
+============
+Alternatives
+============
+
+The zkproof components of Sapling spends and outputs could reasonably be
+construed as authorizing data, rather that information that describes 
+value transfer. As such, it was suggested that these proof values should
+not be committed to by in the transaction identifier, and should instead
+be used as inputs to the signature hash. Proof data is a potential source
+of transaction malleability, and as such this argument is worthy of 
+consideration.
+
+As was stated in :ref:`_requirements-anchor`, a desired property of transaction
+identifiers is that it should be possible for transaction signatures to commit
+to the transaction identifier directly in the case that the transaction contains
+no transparent inputs or in the case that only the ``SIGHASH_ALL`` flag is
+used. Including proofs for signature hash computation but not as part of the
+transaction identifier would make this goal unachievable.
+
+Unlike with transparent use cases where current sources of malleability represent
+a risk of double-spend attacks, in the case of a shielded-only transaction, 
+malleation of a zkproof value would invalidate any signatures associated with
+the transaction, since in the shielded scenario all signatures must commit to 
+all of the transaction's shielded spends, not just a subset, and so it is not
+possible for one party to a transaction to malleate their proofs in a way that
+does not break the signatures of any other party to the transaction. Thus, 
+this source of malleability is not considered to be of concern.
+
+
+==========
+References
+==========
+
+.. [#RFC2119] `RFC 2119: Key words for use in RFCs to Indicate Requirement Levels <https://www.rfc-editor.org/rfc/rfc2119.html>`_
+.. [#zip-0200] `ZIP 200: Network Upgrade Activation Mechanism <https://github.com/zcash/zips/blob/master/zip-0200.rst>`_
+.. [#zip-0076] `ZIP 76: Transaction Signature Validation before Overwinter <https://zips.z.cash/zip-0076>`_
+.. [#zip-0143] `ZIP 143: Transaction Signature Validation for Overwinter <https://zips.z.cash/zip-0143>`_
+.. [#zip-0307] `ZIP 307: Light Client Protocol for Payment Detection <https://zips.z.cash/zip-0307>`_
+
```

Comment:
```suggestion
.. [#zip-0200] `ZIP 200: Network Upgrade Activation Mechanism <zip-0200.rst>`_
.. [#zip-0076] `ZIP 76: Transaction Signature Validation before Overwinter <zip-0076.rst>`_
.. [#zip-0143] `ZIP 143: Transaction Signature Validation for Overwinter <zip-0143.rst>`_
.. [#zip-0307] `ZIP 307: Light Client Protocol for Payment Detection <zip-0307.rst>`_

```
(This URL format works locally, on GitHub, and on zips.z.cash where it's rewritten.)